File: platforms/neoforge-1.21.1/src/main/java/heckerpowered/lethal/platform/services/HostingBlockPositionProvider.java
/*
 * SPDX-License-Identifier: MIT
 * Copyright (c) 2026 heckerpowered
 */

package heckerpowered.lethal.platform.services;

import heckerpowered.bridge.math.BlockPositionProvider;
import heckerpowered.bridge.math.BlockPositionView;
import heckerpowered.lethal.platform.interop.MixinInterop;
import net.minecraft.core.BlockPos;
import org.jetbrains.annotations.NotNull;

public final class HostingBlockPositionProvider implements BlockPositionProvider {
    @Override
    @NotNull
    public BlockPositionView position(int x, int y, int z) {
        return MixinInterop.requireAccess(new BlockPos(x, y, z), BlockPositionView.class);
    }

    @Override
    @NotNull
    public BlockPositionView fromPackedLong(long value) {
        return MixinInterop.requireAccess(BlockPos.of(value), BlockPositionView.class);
    }

    @Override
    public long asLong(@NotNull BlockPositionView position) {
        return BlockPos.asLong(position.getX(), position.getY(), position.getZ());
    }
}

File: platforms/neoforge-1.21.1/src/main/kotlin/heckerpowered/lethal/platform/services/HostingBlockPositionProvider.kt
/*
 * SPDX-License-Identifier: MIT
 * Copyright (c) 2026 heckerpowered
 */

package heckerpowered.lethal.platform.services

import heckerpowered.bridge.math.BlockPositionProvider
import heckerpowered.bridge.math.BlockPositionView
import heckerpowered.lethal.platform.interop.asView
import net.minecraft.core.BlockPos

class HostingBlockPositionProvider : BlockPositionProvider {
    override fun position(x: Int, y: Int, z: Int): BlockPositionView {
        return BlockPos(x, y, z).asView()
    }

    override fun fromPackedLong(value: Long): BlockPositionView {
        return BlockPos.of(value).asView()
    }

    override fun asLong(position: BlockPositionView): Long {
        return BlockPos.asLong(position.x, position.y, position.z)
    }
}
